app/javascript/controllers/available_date_controller.js

```diff
@@ -90,4 +91,15 @@ export default class extends Controller {
       this.errors.day = 'all parts must be provided'
     }
   }
+
+  // Validate that the day selected is in the month
+  validDate() {
+    const day = this.dayTarget.value
+    const month = this.monthTarget.value
+    const year = this.yearTarget.value
+
+    if (isNaN(new Date(`${year}-${month}-${day}`))) {
+      this.errors.day = 'day must exist in selected month'
+    }
```

Comment:
oddly, this seemed to be needed as well;  we only triggered the nil error when we had a user with a nil "last_work_terms_agreement" AND an invalid or incomplete embargo release date.

Reply:
this gives a nice friendly error message to the user on the page if they select a bogus date (e.g. June 31) before they even submit the form